routes/api_v1_0.py

```diff
@@ -915,6 +921,382 @@ def mbt(projection: str, tiletype: str, zoom: int, x: int, y: int):
+    data = DB.session.query(
+        DB.func.min(Station.time),
+        DB.func.max(Station.time),
+    ).filter(Station.platform == platform).one()
+    resp = jsonify({
+        'min': data[0].isoformat(),
+        'max': data[1].isoformat(),
+    })
+    resp.cache_control.max_age = max_age
+    return resp
+
+
+@bp_v1_0.route('/api/v1.0/observation/track/<string:query>.json')
+def observation_track_v1_0(query: str):
+    """
+    API Format: /api/v1.0/observation/track/<string:query>.json
+
+    <string:query> : List of key=value pairs, seperated by ;
+        valid query keys are: start_date, end_date, datatype, platform_type,
+            meta_key, meta_value, mindepth, maxdepth, area, radius, quantum
+
+    Observational query for tracks
+
+    **Used in ObservationSelector**
+    """
+    query_dict = {
+        key: value
+        for key, value in [ q.split('=') for q in query.split(';')]
+    }
+    data = []
+    max_age = 86400
+    params = {}
+
+    MAPPING = {
+        'start_date': 'starttime',
+        'end_date': 'endtime',
+        'platform_type': 'platform_types',
+        'meta_key': 'meta_key',
+        'meta_value': 'meta_value',
+    }
+    for k,v in query_dict.items():
+        if k not in MAPPING:
+            continue
+
+        if k in ['start_date', 'end_date']:
+            params[MAPPING[k]] = dateparse(v)
+        elif k in ['datatype', 'meta_key', 'meta_value']:
+            if k == 'meta_key' and v == 'Any':
+                continue
+            if k == 'meta_value' and query_dict.get('meta_key') == 'Any':
+                continue
+
+            params[MAPPING[k]] = v
+        elif k == 'platform_type':
+            params[MAPPING[k]] = v.split(',')
+        else:
+            params[MAPPING[k]] = float(v)
+
+    if 'area' in query_dict:
+        area = json.loads(query_dict.get('area'))
+        if len(area) > 1:
+            lats = [c[0] for c in area]
+            lons = [c[1] for c in area]
+            params['minlat'] = min(lats)
+            params['minlon'] = min(lons)
+            params['maxlat'] = max(lats)
+            params['maxlon'] = max(lons)
+        else:
+            params['latitude'] = area[0][0]
+            params['longitude'] = area[0][1]
+            params['radius'] = float(query_dict.get('radius', 10))
+
+        platforms = ob_queries.get_platforms(DB.session, **params)
+        for param in [
+            'minlat', 'maxlat', 'minlon', 'maxlon', 'latitude', 'longitude',
+            'radius'
+        ]:
+            if param in params:
+                del params[param]
+
+        params['platforms'] = platforms
+
+    coordinates = ob_queries.get_platform_tracks(
+        DB.session,
+        query_dict.get("quantum", "day"),
+        **params
+    )
+
+    if len(coordinates) == 0:
+        coordinates = [[0, 0, 0, 0]]
+
+    df = pd.DataFrame(np.array(coordinates), columns=['id', 'type', 'lon', 'lat'])
+    df['id'] = df.id.astype(int)
+
+    if len(coordinates) > 1:
```

Comment:
Same thing here `if coordinates`...implicit check.

Reply:
I just removed one check, the other needs to stay as len(coordinates) because it's checking for at least 2 points--no such thing as a track with only one point.